contracts/src/main/kotlin/com/gevamu/contracts/PaymentContract.kt

```diff
@@ -16,21 +16,193 @@
 
 package com.gevamu.contracts
 
+import com.gevamu.states.Payment
 import net.corda.core.contracts.CommandData
+import net.corda.core.contracts.CommandWithParties
 import net.corda.core.contracts.Contract
+import net.corda.core.contracts.ContractState
+import net.corda.core.contracts.TransactionState
+import net.corda.core.internal.Verifier
 import net.corda.core.transactions.LedgerTransaction
+import java.util.UUID
 
 class PaymentContract : Contract {
-    companion object {
-        val ID = "com.gevamu.contracts.PaymentContract"
+    override fun verify(tx: LedgerTransaction) {
+        for (output in tx.outputs) {
+            verifyOutput(tx, output)
+        }
     }
 
-    override fun verify(tx: LedgerTransaction) {
+    private fun verifyOutput(tx: LedgerTransaction, output: TransactionState<ContractState>) {
+        if (output.data !is Payment) {
+            throw IllegalArgumentException("Illegal type of transaction data: ${output.data::class.java.name}")
+        }
+
+        val outputPayment = output.data as Payment
+
+        val command = lookupCommand(tx, outputPayment)
+        when (command.value) {
+            is Commands.Create -> {
+                CreateVerifier(command, outputPayment).verify()
+            }
+            is Commands.SendToGateway -> {
+                val inputPayment = lookupInput(tx, outputPayment)
+                SendToGatewayVerifier(
+                    command = command,
+                    inputPayment = inputPayment,
+                    outputPayment = outputPayment
+                ).verify()
+            }
+            is Commands.UpdateStatus -> {
+                val inputPayment = lookupInput(tx, outputPayment)
+                UpdateStatusVerifier(
+                    command = command,
+                    inputPayment = inputPayment,
+                    outputPayment = outputPayment
+                ).verify()
+            }
+        }
+    }
+
+    private fun lookupCommand(tx: LedgerTransaction, payment: Payment): CommandWithParties<CommandData> {
+        return tx.commands
+            .first {
```

Comment:
I'd throw some more descriptive error if corresponding command isn't found.

Having more than one command with the same uniquePaymentId is also an error.